Dockerfile

```diff
@@ -16,21 +16,37 @@
 
 FROM python:3.6-slim
 
-COPY . /opt/airflow/
-
-ARG AIRFLOW_HOME=/usr/local/airflow
+ENV AIRFLOW_HOME=/usr/local/airflow
 ARG AIRFLOW_DEPS="all"
 ARG PYTHON_DEPS=""
-ARG buildDeps="freetds-dev libkrb5-dev libsasl2-dev libssl-dev libffi-dev libpq-dev git"
-ARG APT_DEPS="$buildDeps libsasl2-dev freetds-bin build-essential default-libmysqlclient-dev apt-utils curl rsync netcat locales"
+ARG BUILD_DEPS="freetds-dev libkrb5-dev libssl-dev libffi-dev libpq-dev git"
+ARG APT_DEPS="libsasl2-dev freetds-bin build-essential default-libmysqlclient-dev apt-utils curl rsync netcat locales"
 
-WORKDIR /opt/airflow
-RUN set -x \
+ENV PATH="$HOME/.npm-packages/bin:$PATH"
+
+RUN set -euxo pipefail \
     && apt update \
     && if [ -n "${APT_DEPS}" ]; then apt install -y $APT_DEPS; fi \
+    && curl -sL https://deb.nodesource.com/setup_10.x | bash - \
+    && apt update \
+    && apt install -y nodejs \
+    && apt autoremove -yqq --purge \
+    && apt clean
+
+COPY . /opt/airflow/
+
+WORKDIR /opt/airflow/airflow/www
+RUN npm install \
+    && npm run prod
```

Comment:
-1 to doing it this way: Having Node and all of the node_modules in the final image isn't great.

Instead we should use a [multi-stage docker file](https://docs.docker.com/develop/develop-images/multistage-build/) so this line becomes:

```
COPY --from=assert-builder /opt/airflow/airflow/www/static/dist /opt/airflow/airflow/www/static/dist
```

And then earlier in the docker file we have (i.e. right at the start)

```
FROM node as asset-builder
COPY ./airflow/www/package*json /opt/airflow/airflow/www/package.json
RUN npm install
COPY ./airflow/www/static/ /opt/airflow/airflow/www/static/
RUN npm run prod
```

Reply:
Same as @jmcarp said in https://github.com/apache/airflow/pull/4724#issuecomment-464546183